src/github.com/getlantern/enproxy/proxy.go

```diff
@@ -169,11 +175,15 @@ func (p *Proxy) handleWrite(resp http.ResponseWriter, req *http.Request, lc *laz
 		badGateway(resp, fmt.Sprintf("Unable to write to connOut: %s", err))
 		return
 	}
-	if p.Host != "" {
-		// Always feed this so clients will be guaranteed to reach
-		// this particular proxy even if they originally reached us
-		// through (e.g.) DNS round robin.
-		resp.Header().Set(X_ENPROXY_PROXY_HOST, p.Host)
+	host := ""
+	if p.HostFn != nil {
+		host = p.HostFn(req)
+	}
+	if host == "" {
+		host = p.Host
+	}
+	if host != "" {
```

Comment:
Cool thanks! I *think* at this point no one else is using enproxy, so we can probably get away with breaking some backwards compatibility, but this looks fine.